src/Hpp/Types.hs

```diff
@@ -85,8 +86,10 @@ instance Functor f => Functor (FreeF f a) where
 
 -- | Dynamic state of the preprocessor engine.
 data HppState = HppState { hppConfig :: Config
+                         , hppCurDir :: FilePath
```

Comment:
This field was added to store the directory of the input. This + the changes above where I add this an extra parameter to `includeCandidates` in the code that looks up include files fix a bug described below:

```
hpp src/Main.hs
```

src/Main.hs:
```
#include "Main2.hs"
```

src/Main2.hs:
```
module Main where

main :: IO ()
main = return ()
```

Without the change I made to the include code, the example I provided above would convert `src/Main.hs` to an empty file or throw an IncludeDoesNotExist error if error values are not ignored. 

The reason is that upon the `#include "Main2.hs"`, `includeCandidates` will search for `Main2.hs` relative to the current directory of `hpp` which is the directory that contains `src`! Hence it won't find `Main2.hs`. I hit this bug when compiling the `vault` package which uses this method of organize code.